### ISSUE TITLE
Nullability flow analysis does not work with logical OR/AND boolean operators.

### ISSUE THREAD

**Issue by angelobreuer** [dotnet/roslyn#59812]
**Version Used**: 
Latest (27 Jan 2022) at 3b32fc36ec4c2a81457d90f965c89c5b6fc37082

**Steps to Reproduce**:

Version with logical OR operator:

```csharp
#nullable enable

A? a = new A();
A? b = new A();

if (a is null | b is null)
{
    return;
}

_ = a.B; // warning CS8602: Dereference of a possibly null reference.

class A { public readonly string B = string.Empty; }
```

https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4GIYFcA2OLA4CmABETAcQLABQtAggPwkQkC8JMRA7ifQBQBKANwNmwdpx58houjSgAzEv1ZQAzp1w4SqEhI1a8g2gG9aJSyQACAdjkBfWrQD6kiADoAQnNrWATHwkpjYAzCQATkQQCAD2MDgAnjYAjAAMJF6S1ukeAKJgAA4ALonCJA5AA==

Version with logical AND operator:

```csharp
#nullable enable

A? a = new A();

if (a is null & true)
{
    return;
}

_ = a.B; // warning CS8602: Dereference of a possibly null reference.

class A { public readonly string B = string.Empty; }
```

https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4GIYFcA2OLA4CmABETAcQLABQtAggPwkQkC8JMRA7ifQBQBKANwNmwdpx58houjSgAzEv1ZQAzp1w4SAMhIAXAE5Yig2gG9aJGyQACAdjkBfWrQD6kiADoAQnNo7ACY+Egt7AGYSIyIIBAB7GBwAT3sARgAGEl9JO0zvAFEwAAcDZOESZyA=

**Expected Behavior**:

Nullability flow analysis should analyze logical boolean operations.

**Actual Behavior**:

CS8602 is reported at `_ = a.B` where `a` (and `a.B`) should be always not-null.

**Comment by jcouv** [comment]
Closing as duplicate of https://github.com/dotnet/roslyn/issues/37344 (which we're using to collect such requests).
